torch/distributed/distributed_c10d.py

```diff
@@ -2081,15 +2113,23 @@ def all_gather(tensor_list, tensor, group=None, async_op=False):
         [tensor([1.+1.j, 2.+2.j]), tensor([3.+3.j, 4.+4.j])] # Rank 0
         [tensor([1.+1.j, 2.+2.j]), tensor([3.+3.j, 4.+4.j])] # Rank 1
 
+    .. warning::
+        Using a single tensor as `output` is not supported by the Gloo backend.
+
     """
-    _check_tensor_list(tensor_list, "tensor_list")
+    # `output` can be a single tensor or a list of tensor.
+    # If it is a single tensor, we route it to the _all_gather_base implementation.
+    if isinstance(output, torch.Tensor):
+        return _all_gather_base(output, tensor, group, async_op)
```

Comment:
To my knowledge, `_all_gather_base` is not a custom op in terms of torch script while `all_gather` is according to https://github.com/pytorch/pytorch/pull/79669.

Could this mean that one function behaves slightly differently depending on the `output`?

Reply:
We can make `_all_gather_base` a custom op in the backend too. That only concerns the internal implementation.